subprojects/core/src/main/java/org/gradle/plugin/use/internal/DefaultPluginRequest.java

```diff
@@ -57,6 +57,17 @@ public String getVersion() {
         return version;
     }
 
+    @Nullable
+    @Override
+    public String getArtifact() {
```

Comment:
It should be a `ModuleVersionSelector`, just like in dependency resolution rules. A String is inconvenient, an Object would leak mutable stuff deeper into our resolver infrastructure.

You can use the `DependencyFactory` to convert from notation to `Dependency` and then create a `DefaultModuleVersionSelector` from that.